discovery-provider/src/tasks/index.py

```diff
@@ -156,31 +157,41 @@ def get_latest_block(db: SessionManager, is_retry: bool = False):
             logger.info(
                 f"index.py | get_latest_block | current={current_block_number} target={target_latest_block_number}"
             )
-            latest_block = update_task.web3.eth.getBlock(
-                target_latest_block_number, True
-            )
-
-            # we've seen potential instances of blocks returning no transactions
-            # if the block had no transactions and this is the first call to the gatewway
-            # retry after small delay to confirm the block is actually empty
-            if len(latest_block.transactions) == 0 and not is_retry:
-                logger.info(
-                    f"index.py | get_latest_block | target={target_latest_block_number} | target block has 0 transactions, retrying to confirm"
-                )
-                time.sleep(0.5)
-                return get_latest_block(db, True)
-
-            # if it retries getting the block and this time it has transactions when it didn't previously
-            if len(latest_block.transactions) > 0 and is_retry:
-                logger.info(
-                    f"index.py | get_latest_block | target={target_latest_block_number} | target block got transactions after retrying, got 0 initially"
-                )
+            latest_block = get_block_with_retry(target_latest_block_number)
 
             return latest_block
     except Exception as e:
         raise Exception(f"index.py | get_latest_block | got exception {e}")
 
 
+def get_block_with_retry(
+    block_identifier: typing.Union[int, str], is_retry: bool = False
```

Comment:
same comment that we should do `retry_count` or similar here so the caller can just specify how much it should retry

 and then check if retry_count > 0